File: jems-server/src/main/kotlin/io/cloudflight/jems/server/project/service/lumpsum/model/ProjectLumpSum.kt
package io.cloudflight.jems.server.project.service.lumpsum.model

import io.cloudflight.jems.server.project.service.model.ProjectPeriod
import java.time.ZonedDateTime

const val PREPARATION_PERIOD_NUMBER = 0
const val CLOSURE_PERIOD_NUMBER = 255

val closurePeriod = ProjectPeriod(
    number = CLOSURE_PERIOD_NUMBER,
    start = -1,
    end = -1
)

data class ProjectLumpSum (
    var orderNr: Int,
    val programmeLumpSumId: Long,
    val period: Int? = null,
    val lumpSumContributions: List<ProjectPartnerLumpSum> = emptyList(),
    val fastTrack: Boolean = false,
    val readyForPayment: Boolean = false,
    val comment: String? = null,
    var paymentEnabledDate: ZonedDateTime? = null,
    var lastApprovedVersionBeforeReadyForPayment: String? = null,
    var installmentsAlreadyCreated: Boolean? = null,
    var linkedToEcPaymentId: Long? = null
)

File: jems-ui/src/app/project/model/lump-sums/projectLumpSum.ts
import {PartnerContribution} from './partnerContribution';

export class ProjectLumpSum {
  orderNr: number;
  programmeLumpSumId: number;
  period: number;
  lumpSumContributions: PartnerContribution[];
  comment: string;
  readyForPayment: boolean;
  fastTrack: boolean;
  paymentEnabledDate: Date;
  lastApprovedVersionBeforeReadyForPayment: string;
  installmentsAlreadyCreated: boolean;
  linkedToEcPaymentId: number;

  constructor(
    orderNr: number,
    programmeLumpSumId: number,
    period: number,
    lumpSumContributions: PartnerContribution[],
    comment: string,
    readyForPayment: boolean,
    fastTrack: boolean,
    paymentEnabledDate: Date,
    lastApprovedVersionBeforeReadyForPayment: string,
    installmentsAlreadyCreated: boolean,
    linkedToEcPaymentId: number
  ) {
    this.orderNr = orderNr;
    this.programmeLumpSumId = programmeLumpSumId;
    this.period = period;
    this.lumpSumContributions = lumpSumContributions;
    this.comment = comment;
    this.readyForPayment = readyForPayment;
    this.fastTrack = fastTrack;
    this.paymentEnabledDate = paymentEnabledDate;
    this.lastApprovedVersionBeforeReadyForPayment = lastApprovedVersionBeforeReadyForPayment;
    this.installmentsAlreadyCreated = installmentsAlreadyCreated;
    this.linkedToEcPaymentId = linkedToEcPaymentId;
  }
}
